courses-app/src/components/Courses/__tests__/Courses.test.js

```diff
@@ -0,0 +1,108 @@
+import { Provider } from 'react-redux';
+import * as reactRedux from 'react-redux';
+import { BrowserRouter, useHistory } from 'react-router-dom';
+import { render, fireEvent, screen } from '@testing-library/react';
+
+import Courses from '../Courses';
+import * as selectors from '../../../store/selectors';
+import { mockedAuthorsList, mockedCoursesList } from '../../../constants';
+
+jest.mock('react-router-dom', () => ({
+	...jest.requireActual('react-router-dom'),
+	useHistory: jest.fn(),
+}));
+
+jest.mock('../../../store/selectors', () => ({
+	getCourses: jest.fn(),
+	getAuthors: jest.fn(),
+	getUser: jest.fn(),
+}));
+
+jest.spyOn(reactRedux, 'useDispatch');
+
+const mockedState = {
+	user: {
+		isAuth: true,
+		name: 'Test Name',
+		token: 'ddd32432-34324sfs-few2113',
+		role: 'admin',
+		email: 'test@test.ru',
+	},
+	courses: mockedCoursesList,
+	authors: mockedAuthorsList,
+};
+
+const mockedStore = {
+	getState: () => mockedState,
+	subscribe: jest.fn(),
+	dispatch: jest.fn(),
+};
+
+beforeEach(() => {
+	selectors.getAuthors.mockReturnValue(mockedState.authors);
+	selectors.getCourses.mockReturnValue(mockedState.courses);
+	selectors.getUser.mockReturnValue(mockedState.user);
+});
+
+afterEach(() => {
+	selectors.getAuthors.mockClear();
+	selectors.getCourses.mockClear();
+	selectors.getUser.mockClear();
+});
+
+describe('Courses', () => {
+	it('Courses rendered', () => {
+		const { container } = render(
+			<Provider store={mockedStore}>
+				<BrowserRouter>
+					<Courses />
+				</BrowserRouter>
+			</Provider>
+		);

```

Comment:
this part could be as bar of the beforeEach block in every test file - it will reduce quantity of the lines